src/include/optimizer/rule_rewrite.h

```diff
@@ -0,0 +1,122 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// rule_rewrite.h
+//
+// Identification: src/include/optimizer/rule_rewrite.h
+//
+// Copyright (c) 2015-16, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "optimizer/rule.h"
+#include "optimizer/absexpr_expression.h"
+
+#include <memory>
+
+namespace peloton {
+namespace optimizer {
+
+using GroupExprTemplate = GroupExpression;
+using OptimizeContext = OptimizeContext;
+
+/* Rules are applied from high to low priority */
+enum class RulePriority : int {
+  HIGH = 3,
+  MEDIUM = 2,
+  LOW = 1
+};
+
```

Comment:
This seems like a good place to use macros. All the header stubs look the same.

Reply:
Resolving.